app/models/statistics_announcement.rb

```diff
@@ -22,7 +22,7 @@ def can_publish_to_publishing_api?
 
   belongs_to :creator, class_name: "User"
   belongs_to :cancelled_by, class_name: "User"
-  belongs_to :publication
+  belongs_to :publication, autosave: true
```

Comment:
Oh, interesting 🤔 -> is this needed so that the `before_validation` runs?

Reply:
no - this is needed because Rubocop doesn't allow publication.save, only publication.save! - and that's in the middle of a transaction, so we don't really want it to be called there, as there could be other validation issues with the publication (e.g. unmodifiable states). 
autosave will also save the publication after saving the announcement and all other validations have been run, so it's at a more suitable point to save.